src/sleep_dataset.py

```
@@ -0,0 +1,106 @@
+    "presence of specific waveforms). Then, based on your analysis, classify the "
+    "sleep stage.\n\nAnswer: "
+)
+
+
+def _download_csv():
+    """Download the Sleep-EDF CoT CSV if not already cached."""
+    if os.path.exists(DATA_PATH):
+        return
+    os.makedirs(DATA_DIR, exist_ok=True)
+    print(f"Downloading Sleep-EDF CoT data to {DATA_PATH}...")
+    urllib.request.urlretrieve(DATA_URL, DATA_PATH)
```

Ruff (0.15.6)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)